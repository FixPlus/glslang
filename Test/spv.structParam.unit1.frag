#version 460 core

struct MyStruct {
    vec3 v;
    int i;
};

int foo(inout MyStruct s)
{
    s.v = vec3(0.f, 1.f, 2.f);
    return s.i;
}
