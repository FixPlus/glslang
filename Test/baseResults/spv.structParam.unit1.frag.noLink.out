spv.structParam.unit1.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:8  Function Definition: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:8    Function Parameters: 
0:8      's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10    Sequence
0:10      move second child to first child ( temp highp 3-component vector of float)
0:10        v: direct index for structure ( global highp 3-component vector of float)
0:10          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10          Constant:
0:10            0 (const int)
0:10        Constant:
0:10          0.000000
0:10          1.000000
0:10          2.000000
0:11      Branch: Return with expression
0:11        i: direct index for structure ( global highp int)
0:11          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:11          Constant:
0:11            1 (const int)
0:?   Linker Objects
0:?     Linker Function: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( inout structure{ global highp 3-component vector of float v,  global highp int i})

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 25

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              Source GLSL 460
                              Name 5  "MyStruct"
                              MemberName 5(MyStruct) 0  "v"
                              MemberName 5(MyStruct) 1  "i"
                              Name 9  "foo(struct-MyStruct-vf3-i11;"
                              Name 8  "s"
                              Decorate 9(foo(struct-MyStruct-vf3-i11;) Linkage Attributes 678391654 1970435187 1294824547 1920226169 762602357 758343286 993079657 0 0
               2:             TypeFloat 32
               3:             TypeVector 2(float) 3
               4:             TypeInt 32 1
     5(MyStruct):             TypeStruct 3(fvec3) 4(int)
               6:             TypePointer Function 5(MyStruct)
               7:             TypeFunction 4(int) 6(ptr)
              11:      4(int) Constant 0
              12:    2(float) Constant 0
              13:    2(float) Constant 1065353216
              14:    2(float) Constant 1073741824
              15:    3(fvec3) ConstantComposite 12 13 14
              16:             TypePointer Function 3(fvec3)
              18:      4(int) Constant 1
              19:             TypePointer Function 4(int)
              23:             TypeVoid
9(foo(struct-MyStruct-vf3-i11;):      4(int) Function None 7
            8(s):      6(ptr) FunctionParameter
              10:             Label
              17:     16(ptr) AccessChain 8(s) 11
                              Store 17 15
              20:     19(ptr) AccessChain 8(s) 18
              21:      4(int) Load 20
                              ReturnValue 21
                              FunctionEnd
